ui/panels-plugin/src/plugins/time-series-chart/utils/palette-gen.test.ts

```diff
@@ -11,26 +11,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { getSeriesColor } from './palette-gen';
+import { getAutoPaletteColor, getCategoricalPaletteColor, getConsistentSeriesNameColor } from './palette-gen';
 
-describe('getSeriesColor', () => {
-  it('should return Auto generated color', () => {
-    const value = getSeriesColor('p90 test api subdomain', 2, ['#fff', '000', '#111', '#222', '#333'], 'Auto');
-    expect(value).toEqual('hsla(1008901844,50%,50%,0.8)');
-  });
+describe('getCategoricalPaletteColor', () => {
+  const fallbackColor = '#ff0000';
 
   it('should return 1st color in Categorical palette', () => {
-    const value = getSeriesColor('p90 test api subdomain', 0, ['#fff', '000', '#111', '#222', '#333'], 'Categorical');
-    expect(value).toEqual('#fff');
+    const paletteColor = getCategoricalPaletteColor(['#fff', '000', '#111', '#222', '#333'], 0, fallbackColor);
+    expect(paletteColor).toEqual('#fff');
   });
 
   it('should return 3rd color in Categorical palette', () => {
-    const value = getSeriesColor('p90 test api subdomain', 2, ['#fff', '000', '#111', '#222', '#333'], 'Categorical');
-    expect(value).toEqual('#111');
+    const paletteColor = getCategoricalPaletteColor(['#fff', '000', '#111', '#222', '#333'], 2, fallbackColor);
+    expect(paletteColor).toEqual('#111');
+  });
+
+  it('should repeat color after looping through entire palette', () => {
+    const paletteColor = getCategoricalPaletteColor(['#fff', '000', '#111', '#222', '#333'], 5, fallbackColor);
+    expect(paletteColor).toEqual('#fff');
   });
+});
```

Comment:
Could add a test for when it returns `fallback` color? (if the array is undefined, I'm guessing)

Reply:
<3 it